modules/decima-ext-model-exporter/src/main/java/com/shade/decima/ui/data/viewer/model/BaseModelExporter.java

```diff
@@ -5,67 +5,64 @@
 import com.shade.decima.model.rtti.objects.RTTIReference;
 import com.shade.decima.model.rtti.registry.RTTITypeRegistry;
 import com.shade.decima.model.rtti.types.RTTITypeEnum;
-import com.shade.decima.ui.data.viewer.model.utils.Matrix3x3;
-import com.shade.decima.ui.data.viewer.model.utils.Matrix4x4;
-import com.shade.decima.ui.data.viewer.model.utils.Transform;
-import com.shade.decima.ui.data.viewer.model.utils.Vector3;
 import com.shade.platform.model.util.MathUtils;
 import com.shade.util.NotNull;
+import org.joml.*;
 
 import java.util.UUID;
 
 public class BaseModelExporter {
     @NotNull
-    protected static Transform worldTransformToTransform(@NotNull RTTIObject transform) {
+    protected static Matrix4dc worldTransformToMatrix(@NotNull RTTIObject transform) {
         final var pos = transform.obj("Position");
         final var ori = transform.obj("Orientation");
         final var col0 = ori.obj("Col0");
         final var col1 = ori.obj("Col1");
         final var col2 = ori.obj("Col2");
-
-        final Vector3 translation = new Vector3(new double[]{
+        final Matrix3dc tmpMatrix = new Matrix3d(
+            col0.f32("X"), col0.f32("Y"), col0.f32("Z"),
+            col1.f32("X"), col1.f32("Y"), col1.f32("Z"),
+            col2.f32("X"), col2.f32("Y"), col2.f32("Z")
+        ).transpose();
+        final Matrix4dc rotationMatrix = new Matrix4d(tmpMatrix);
+        Vector3d translation = new Vector3d(
             pos.f64("X"),
             pos.f64("Y"),
             pos.f64("Z")
-        });
-
-        final Matrix3x3 rotationAndScale = new Matrix3x3(new double[][]{
-            new double[]{col0.f32("X"), col0.f32("Y"), col0.f32("Z")},
-            new double[]{col1.f32("X"), col1.f32("Y"), col1.f32("Z")},
-            new double[]{col2.f32("X"), col2.f32("Y"), col2.f32("Z")}
-        });
-
-        return new Transform(translation, rotationAndScale.toQuaternion(), rotationAndScale.toScale());
+        );
+        final Matrix4d translationMatrix = new Matrix4d().translate(translation);
+        translationMatrix.mul(rotationMatrix, translationMatrix);
+        return translationMatrix;
```

Comment:
I belive it can be narrowed down to
```java
return new Matrix4d(
    col0.f32("X"), col0.f32("Y"), col0.f32("Z"), pos.f64("X"),
    col1.f32("X"), col1.f32("Y"), col1.f32("Z"), pos.f64("Y"),
    col2.f32("X"), col2.f32("Y"), col2.f32("Z"), pos.f64("Z"),
    0, 0, 0, 1
);
```